process/block/shardblock_test.go

```diff
@@ -135,193 +136,88 @@ func CreateMockArgumentsMultiShard(
 
 // ------- NewBlockProcessor
 
-func TestNewBlockProcessor_NilDataPoolShouldErr(t *testing.T) {
+func TestNewShardProcessor(t *testing.T) {
 	t.Parallel()
 
 	coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
-	dataComponents.DataPool = nil
-	arguments := CreateMockArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
-	sp, err := blproc.NewShardProcessor(arguments)
-
-	assert.Equal(t, process.ErrNilDataPoolHolder, err)
-	assert.Nil(t, sp)
-}
 
-func TestNewBlockProcessor_NilHeadersDataPoolShouldErr(t *testing.T) {
-	t.Parallel()
+	tests := []struct {
+		args        func() blproc.ArgShardProcessor
+		expectedErr error
+	}{
+		{
+			args: func() blproc.ArgShardProcessor {
+				return CreateMockArgumentsMultiShard(coreComponents, nil, bootstrapComponents, statusComponents)
+			},
+			expectedErr: process.ErrNilDataComponentsHolder,
+		},
+		{
+			args: func() blproc.ArgShardProcessor {
```

Comment:
almost the same number of lines of code as t.Run :D
Can be left as it is

Reply:
Will leave it like this then